lib/pages/first_setup/widgets/first_setup_bloc.dart

```diff
@@ -0,0 +1,30 @@
+import 'package:blueboard/pages/first_setup/widgets/first_setup_event.dart';
+import 'package:blueboard/pages/first_setup/widgets/first_setup_state.dart';
+import 'package:blueboard/providers/user_provider.dart';
+import 'package:blueboard/services/navigation.dart';
+import 'package:bloc/bloc.dart';
+import 'package:dio/dio.dart';
+
+class FirstSetupBloc extends Bloc<FirstSetupEvent, FirstSetupState> {
+	@override
+	get initialState => FirstSetupState.initial();
+
+	@override
+	Stream<FirstSetupState> mapEventToState(FirstSetupEvent event) async* {
+		if (event is SaveUserData) {
+			yield* _saveProfile(firstName: event.firstName, lastName: event.lastName);
+		}
+	}
+
+	Stream<FirstSetupState> _saveProfile({String firstName, String lastName}) async* {
+		yield FirstSetupState.loading();
+
+		try {
+			await UserProvider().initialUpdate(firstName, lastName);
```

Comment:
Move user provider to final field (Do not use UserProvider(), instead use _userProvider field)